src/core.jl

```diff
@@ -0,0 +1,180 @@
+"""
+Type representing any object drawable on image
+"""
+abstract Drawable
+
+"""
+    p = Point(x,y)
+    p = Point(c)
+
+A `Drawable` point on the image
+"""
+immutable Point <: Drawable
+    x::Int
+    y::Int
+end
+
+abstract Line <: Drawable
+abstract Circle <: Drawable
+
+
+"""
+    line = LineTwoPoints(p1, p2)
+
+A `Drawable` infinite length line passing through the two points
+`p1` and `p2`.
+"""
+immutable LineTwoPoints <: Line
+    p1::Point
+    p2::Point
+end
+
+"""
+    line = LineNormal(ρ, θ)
+
+A `Drawable` infinte length line having perpendicular length `ρ` from
+origin and angle `θ` between the perpendicular and x-axis
+
+"""
+immutable LineNormal{T<:Real, U<:Real} <: Line
+    ρ::T
+    θ::U
+end
+
+"""
+    circle = CircleThreePoints(p1, p2, p3)
+
+A `Drawable` circle passing through points `p1`, `p2` and `p3`
+"""
+immutable CircleThreePoints <: Circle
+    p1::Point
+    p2::Point
+    p3::Point
+end
+
+"""
+    circle = CirclePointRadius(center, ρ)
+
+A `Drawable` circle having center `center` and radius `ρ`
+"""
+immutable CirclePointRadius{T<:Real} <: Circle
+    center::Point
+    ρ::T
+end
+
+"""
+    ls = LineSegment(p1, p2)
+
+A `Drawable` finite length line between `p1` and `p2`
+"""
+immutable LineSegment <: Drawable
+    p1::Point
+    p2::Point
+end
+
+"""
+    path = Path([point])
+
+A `Drawable` sequence of line segments connecting consecutive pairs
+of points in `[point]`.
+!!! note
+    This will create a non-closed path. For a closed path, see `Polygon`
+"""
+immutable Path <: Drawable
+    vertices::Vector{Point}
+end
+
+"""
+    ellipse = Ellipse(center, ρx, ρy)
+
+A `Drawable` ellipse with center `center` and parameters `ρx` and `ρy`
+
+"""
+immutable Ellipse{T<:Real, U<:Real} <: Drawable
+    center::Point
+    ρx::T
+    ρy::U
+end
+
+"""
+    polygon = Polygon([vertex])
+
+A `Drawable` polygon i.e. a closed path created by joining the
+consecutive points in `[vertex]` along with the first and last point.
+!!! note
```

Comment:
Does this doc-string render properly in the REPL?  I think there needs to be a newline above this line